### ISSUE TITLE
Authelia won't redirect properly

### ISSUE THREAD

**Issue by QuarkZ26** [authelia/authelia#3343]
Hello,

I have tried to install Authelia for both Nginx and Caddy but always end up with the same error:

> level=info msg="Access to https://domain.duckdns.org/agent.ashx (method GET) is not authorized to user anonymous, responding with status code 401 with location redirect to https://authdomain.duckdns.org?rd=https%3A%2F%2Fdomain.duckdns.org%2Fagent.ashx&rm=GET" method=GET path=/api/verify remote_ip=10.1.1.1

I have tested with different apps, some being docker containers and some installed natively. I don't understand why it's not working at this point, any help would be appreciated.

 There are also <> around the word anonymous, but it wouldn't show here.

**Comment by james-d-elliott** [comment]
That's not an error. Are you using official examples (https://www.authelia.com/docs/deployment/supported-proxies/) or are you deviating from them, or are you doing something else entirely?

For nginx you shouldn't get the location redirect part if configured correctly.

**Comment by QuarkZ26** [comment]
I did follow the basic example for Caddy. 

For Nginx I actually used the help of two different videos regarding the setup but each time it won't work and return the same message, this is what pushed me to try Caddy.

I get to the login page and login successfully but then the page refreshes, throws that message and goes back to Authelia. I don't really understand what the message means and why I get it on different configurations/apps

**Comment by nightah** [comment]
I believe this could be related to the fact that your `session.domain` is set to `duckdns.org` can you please confirm?

From what I recall you cannot set cookies for said root level domain, I'm not entirely sure if it even works for sub-domains on the said root but you can probably confirm that pretty easily.

**Comment by QuarkZ26** [comment]
> I believe this could be related to the fact that your `session.domain` is set to `duckdns.org` can you please confirm?
> 
> From what I recall you cannot set cookies for said root level domain, I'm not entirely sure if it even works for sub-domains on the said root but you can probably confirm that pretty easily.

That is correct. What should I setup instead, or is it not possible to do that at all with duckdns?

**Comment by nightah** [comment]
I believe this works with duckdns however, you need to have your `session.domain` set to a subdomain of the root so for example:

`quark.duckdns.org`

Then all of your domains should reside as a subdomain of `quark.duckdns.org` like `auth.quark.duckdns.org`.

**Comment by QuarkZ26** [comment]
> I believe this works with duckdns however, you need to have your `session.domain` set to a subdomain of the root so for example:
> 
> `quark.duckdns.org`
> 
> Then all of your domains should reside as a subdomain of `quark.duckdns.org` like `auth.quark.duckdns.org`.

That was it! wow, I have been trying to figure this out for a while. 

At first I made the change on the forward only and it did the same thing, then I also updated the Authelia subdomain and it started working. Thank you so much everyone!

Just an FYI, is someone else ever encounter this issue, duckdns doesn't allow you to sub-subdomain, however it does use wildcards, so if you use domain.duckdns.org, it'll forward anything that goes sub.domain.duckdns.org
